src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PublicWorkerCraftingRequestResolver.java

```diff
@@ -76,7 +75,12 @@ public void onRequestedRequestCancelled(@NotNull final IRequestManager manager,
     @Override
     public ITextComponent getRequesterDisplayName(@NotNull final IRequestManager manager, @NotNull final IRequest<?> request)
     {
-        return new StringTextComponent(((AbstractBuildingCrafter.View) manager.getColony().getRequesterBuildingForPosition(getLocation().getInDimensionLocation())).getJobName());
+        final IRequester requester = manager.getColony().getRequesterBuildingForPosition(getLocation().getInDimensionLocation());
+        if (requester instanceof AbstractBuildingCrafter.View)
+        {
+            return new StringTextComponent(((AbstractBuildingCrafter.View) requester).getJobName());
```

Comment:
Do we know how many other types there are?

Reply:
too many, this will be fixed for the other ones post-building rework

Reply:
Ok. If that's going to be a while, it might make sense to do an interface for this, that we propagate as needed.